packages/sycamore-macro/src/component.rs

```diff
@@ -219,28 +221,88 @@ impl ToTokens for ComponentFn {
     }
 }
 
+pub struct ParenthesizedTokens {
+    paren: Paren,
+    tokens: TokenStream,
+}
+
+impl ToTokens for ParenthesizedTokens {
+    fn to_tokens(&self, out: &mut TokenStream) {
+        let ParenthesizedTokens { paren, tokens } = self;
+        paren.surround(out, |tout| tout.extend(tokens.clone()));
+    }
+}
+
+impl Parse for ParenthesizedTokens {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let content;
+        let paren = parenthesized!(content in input);
+        let tokens = content.parse()?;
+        Ok(Self { paren, tokens })
+    }
+}
+
+pub struct ComponentAttrArgs {
+    ident: Ident,
+    call: ParenthesizedTokens,
+}
```

Comment:
Now that I look at this, I think this struct should be replaced with just [`syn::Meta`](https://docs.rs/syn/latest/syn/enum.Meta.html) which is also more general since it also allows `#[path]` and `#[attr = value]`

Reply:
Oh ye, i only thought of valid rust macros but other macros applied to the struct means that those macros may require other attribute styles